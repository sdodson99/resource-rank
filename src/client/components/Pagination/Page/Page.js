import React from 'react';
import PropTypes from 'prop-types';
import styles from './Page.module.css';

const Page = ({ number, onClick, isSelected }) => {
  const onPageClick = () => onClick && onClick(number);

  const calculateClassName = () => {
    if (isSelected) {
      return styles.SelectedPage;
    }

    return styles.Page;
  };

  return (
    <button
      className={calculateClassName()}
      data-testid="Page"
      type="button"
      onClick={onPageClick}
      disabled={isSelected}
    >
      {number}
    </button>
  );
};

Page.propTypes = {
  number: PropTypes.number,
  onClick: PropTypes.func,
  isSelected: PropTypes.bool,
};

Page.defaultProps = {
  number: 1,
};

export default Page;
