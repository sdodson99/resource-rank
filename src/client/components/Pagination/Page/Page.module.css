.Page {
  @apply px-3 cursor-pointer text-blue-600 hover:text-blue-800 hover:underline transition duration-200;
}

.SelectedPage {
  @apply px-3 cursor-default;
}
