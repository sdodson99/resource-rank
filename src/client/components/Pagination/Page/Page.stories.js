import React from 'react';
import Page from './Page';

export default {
  component: Page,
  title: 'Components/Pagination/Page',
};

const Template = (args) => <Page {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Number = Template.bind({});
Number.args = {
  number: 3,
};

export const Selected = Template.bind({});
Selected.args = {
  number: 3,
  isSelected: true,
};
