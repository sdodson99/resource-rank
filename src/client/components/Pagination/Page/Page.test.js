import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import Page from './Page';

describe('<Page />', () => {
  it('should mount', () => {
    render(<Page />);

    const page = screen.getByTestId('Page');

    expect(page).toBeInTheDocument();
  });

  it('should execute on click with page number when clicked', () => {
    const mockOnClick = jest.fn();
    render(<Page number={5} onClick={mockOnClick} />);
    const page = screen.getByTestId('Page');
    
    page.click();

    expect(mockOnClick).toBeCalledWith(5);
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<Page number={3} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly when selected', () => {
    const component = createRenderer().render(<Page isSelected={true} />);

    expect(component).toMatchSnapshot();
  });
});
