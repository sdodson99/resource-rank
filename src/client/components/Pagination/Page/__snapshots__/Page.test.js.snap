// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render correctly 1`] = `
<button
  className="Page"
  data-testid="Page"
  onClick={[Function]}
  type="button"
>
  3
</button>
`;

exports[`<Page /> should render correctly when selected 1`] = `
<button
  className="SelectedPage"
  data-testid="Page"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
